flake.nix

```diff
@@ -34,7 +43,22 @@
       packages = forAllSystems (system: pkgs: {
         default = self.packages.${system}.${projectName};
         ${projectName} = pkgs.poetry2nix.mkPoetryApplication {
-          projectDir = ./.;
+          projectDir = ./.; 
+          overrides = pkgs.poetry2nix.defaultPoetryOverrides.extend
+          (self: super: {
+            types-mock = super.types-mock.overridePythonAttrs
+            (
+              old: {
+                buildInputs = (old.buildInputs or [ ]) ++ [ super.setuptools ];
+              }
+            );
+            cashu = super.cashu.overridePythonAttrs (
+              old: {
+                nativeBuildInputs = (old.nativeBuildInputs or [ ]) ++ [ (import legacy-nixpkgs {inherit system;}).python3Packages.setuptools ];
```

Comment:
Let's not deal with this here, but let's fix cashu upstream and then use the fixed version of the upstream.